x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/SqlProtocolTestCase.java

```diff
@@ -252,14 +253,13 @@ private void assertQuery(String sql, String columnName, String columnType, Objec
 
         // randomize binary response enforcement for drivers (ODBC/JDBC) and CLI
         boolean binaryCommunication = randomBoolean();
-        Mode m = Mode.fromString(mode);
         if (randomBoolean()) {
             // set it explicitly or leave the default (null) as is
             requestContent = new StringBuilder(requestContent)
                     .insert(requestContent.length() - 1, ",\"binary_format\":" + binaryCommunication).toString();
-            binaryCommunication = ((Mode.isDriver(m) || m == Mode.CLI) && binaryCommunication);
+            binaryCommunication = ((Mode.isDriver(mode) || Mode.isCli(mode)) && binaryCommunication);
```

Comment:
`Mode.isCli` doesn't add much value and since `Mode.isDriver() || Mode.isCli` is a common pattern, it's worth wrapping that into its own method: `Mode.isInternalClient` or `Mode.isDedicatedClient` or `Mode.isSqlClient` to differentiate between our "own" clients and everything else.

Reply:
Agreed, fixed it.